openquake/hazardlib/calc/conditioned_gmfs.py

```diff
@@ -329,295 +353,300 @@ def get_conditioned_mean_and_covariance(
         dict(truncation_level=0, imtls=observed_imtls,
              maximum_distance=maximum_distance))
 
+    # Generate the contexts and calculate the means and 
+    # standard deviations at the *target* sites ("_Y")
+    cmaker_Y = ContextMaker(
+        rupture.tectonic_region_type, [gsim], dict(
+            truncation_level=0, imtls={target_imts[0].string: [0]},
+            maximum_distance=maximum_distance))
+
     gc_D = GmfComputer(rupture, station_sitecol, cmaker_D)
```

Comment:
Perhaps we don't even need to instantiate a GmfComputer here and in the next line; we only need the appropriate rupture contexts. So we could replace this with:

```python
ctx_D = cmaker_D.get_rctx(rupture.rupture, sitecol)
```

and replace the `gc_D.ctx` that appears further on with `ctx_D`

Reply:
Done